Logger/FileLogger.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+namespace Logger
+{
+    public class FileLogger : BaseLogger
+    {
+        private string path;
+
+        public FileLogger(string path)
+        {
+            
+            //check if the path is valid
+
+            if (File.Exists(path)){
+                Console.WriteLine("the " + path + " is already existing");
+            }
+
+            else
+            {
+                Console.WriteLine("path doesn't exist, enter your path here: ");
+                path = Console.ReadLine();
+                File.Create(path);
+                
+            } //end check
+
+            //update new path...
+            this.path = path;
+
+        }
+
+        public override void Log(LogLevel logLevel, string message)
+        {
+            StreamWriter streamWriter = new StreamWriter(this.path);
+            streamWriter.WriteLine( DateTime.Now.ToString() + " " + className + " " + logLevel + " " + message);
```

Comment:
This doesn't quite work. This will always only have a single line in the output file since this [overload of the StreamWriter](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=netcore-3.0#System_IO_StreamWriter__ctor_System_String_) will overwrite the existing file. There is an [overload that takes in bool](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=netcore-3.0#System_IO_StreamWriter__ctor_System_String_System_Boolean_) that will append to the file rather than overwriting it. This is the reason for your the unit test failure with the message: "System.IO.IOException: The process cannot access the file"
String interpolation would have also made this a lot cleaner.
```suggestion
            streamWriter.WriteLine( $"{DateTime.Now} {className} {logLevel} {message}");
```